source/Irrlicht/CBAWMeshWriter.h

```diff
@@ -21,11 +23,42 @@ namespace scene
 
 	class CBAWMeshWriter : public IMeshWriter
 	{
+	public:
+		enum E_COMPRESSION_TARGETS
+		{
+			ECT_NOTHING = 0x00,
+			ECT_RAW_BUFFERS = 0x01,
+			ECT_TEXTURES = 0x02,
+			ECT_MESH_BUFFERS = 0x04,
+			ECT_DATA_FORMAT_DESC = 0x08,
+			ECT_ANIMATION_DATA = 0x10,
+			ECT_MESHES = 0x20,
+			ECT_EVERYTHING = 0xffffffffu
+		};
+
+		struct WriteProperties
+		{
+			WriteProperties() : blobLz4EncrThresh(4096), blobLzmaEncrThresh(32768), encryptBlobBitField(ECT_RAW_BUFFERS) {}
+			size_t blobLz4EncrThresh;
+			size_t blobLzmaEncrThresh;
+			char encryptionPassPhrase[16];
+			char initializationVector[16];
+			uint64_t encryptBlobBitField;
+		};
+
 	private:
 		struct SContext
 		{
 			core::array<core::BlobHeaderV0> headers;
 			core::array<uint32_t> offsets;
+			const WriteProperties* props;
+		};
+		struct LzmaMemMngmnt
```

Comment:
Don't have this in two places.
I know its in separate classes, but still D.R.Y. unless you have a reason for actual different implementations.

Reply:
Without this there'll be even more repeating (explicitly passing treshold values to writing function). Besides I don't really see repeating in this.

Reply:
you have one LzmaMemMngmnt for write and one LzmaMemMngmnt for load ?

Should't this be in TypedBlob or CBAWFile?

Reply:
Oh.. I thought you were talking about props pointer :D
Sure, I just forgot about this. I'll move it to CBawFile.h